spring-kafka/src/main/java/org/springframework/kafka/listener/KafkaMessageListenerContainer.java

```diff
@@ -534,6 +535,9 @@ public void run() {
 			if (this.genericListener instanceof ConsumerSeekAware) {
 				((ConsumerSeekAware) this.genericListener).registerSeekCallback(this);
 			}
+			if (this.transactionManager != null) {
+				ProducerFactoryUtils.setContainerGroupId(this.consumerGroupId);
```

Comment:
Why do you call it as `container` and use `consumer`? 
Looks like inconsistency and may confuse end-user since containers in the same group may have different consumer group.